zeus/device/cpu/intel.py

```diff
@@ -0,0 +1,151 @@
+    logger.info("RAPL is available.")
+    return True
+
+class ZeusRAPLNotSupportedError(ZeusBaseCPUError):
+    """Zeus CPU exception class Wrapper for RAPL not supported on CPU."""
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+class ZeusRAPLFileInitError(ZeusBaseCPUError):
+    """Zeus CPU exception class Wrapper for RAPL file initialization error on CPU."""
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+class RAPLFile:
+    """RAPL File class for each RAPL file.
+
+    This class defines the interface for interacting with a RAPL file for a package. A package can
+    be a CPU or DRAM
+    """
+
+    def __init__(self, path: str) -> None:
+        self.path: str = path
+        self.energy_uj_path: str = os.path.join(path, "energy_uj")
+        try:
+            with open(os.path.join(path, "name"), 'r') as name_file:
+                self.name: str = name_file.read().strip()
+        except:
+            raise ZeusRAPLFileInitError(
+                "Error reading package name"
+            )
+        try:
+            with open(self.energy_uj_path) as energy_file:
+                self.last_energy: int = int(float(energy_file.read().strip())/10**3)
+        except:
+            raise ZeusRAPLFileInitError(
+                "Error reading package energy"
+            )
+        try:
+            with open(os.path.join(path, "max_energy_range_uj"), 'r') as max_energy_file:
+                self.max_energy_range_uj: int = int(float(max_energy_file.read().strip())/10**3)
+        except:
+            raise ZeusRAPLFileInitError(
+                "Error reading package max energy range"
+            )
+
+    def __str__(self) -> str:
+        return f"Path: {self.path}\nEnergy_uj_path: {self.energy_uj_path}\nName: {self.name}\
+        \nLast_energy: {self.last_energy}\nMax_energy: {self.max_energy_range_uj}"
+
+    def read(self) -> int:
+        with open(self.energy_uj_path) as energy_file:
+            self.last_energy = int(float(energy_file.read().strip())/10**3)
+        return self.last_energy
+
+    def read_delta(self) -> int:
+        last_energy: int = self.last_energy
+        new_energy: int = self.read()
+        if new_energy < last_energy:
+            return new_energy + self.max_energy_range_uj - last_energy
+        return new_energy - last_energy
+
+class INTELCPU(cpu_common.CPU):
+    """Control a Single Intel CPU using RAPL interface."""
+
+    def __init__(self, cpu_index: int) -> None:
+        """Initialize the Intel CPU with a specified index."""
+        super().__init__(cpu_index)
+        self._get_powerzone()
+
+    _exception_map = {
+        FileNotFoundError: cpu_common.ZeusCPUNotFoundError,
+        PermissionError: cpu_common.ZeusCPUNoPermissionError,
+        OSError: cpu_common.ZeusCPUInitError,
+    }
+
+    def _get_powerzone(self) -> None:
```

Comment:
Please have a block of docstring explaining RAPL terms like RAPL, powerzone, package, etc. in `zeus/device/cpu/intel.py` (or if we rename it, `rapl.py`).